src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/admin/AdminDeleteSessionTemplateTest.kt

```diff
@@ -63,4 +64,24 @@ class AdminDeleteSessionTemplateTest(
     Assertions.assertThat(testSessionLocationGroupRepository.hasJoinTable(sessionTemplate.id, sessionGroup1.id)).isFalse
     Assertions.assertThat(testSessionLocationGroupRepository.hasJoinTable(sessionTemplate.id, sessionGroup2.id)).isFalse
   }
+
+  @Test
+  fun `cannot delete session template by reference with existing visits test`() {
```

Comment:
could add another test for non booked visits post adding status on query

Reply:
resolved after having a chat with @AledWEvans as this cannot be done once the session template goes Live and is used to create visits whatever the Status.